lib/block_view.go

```diff
@@ -4165,6 +4290,73 @@ func (bav *UtxoView) _deleteProfileEntryMappings(profileEntry *ProfileEntry) {
 	bav._setProfileEntryMappings(&tombstoneProfileEntry)
 }
 
+// Takes a Postgres Profile, sets all the mappings on the view, returns the equivalent ProfileEntry and PKIDEntry
+func (bav *UtxoView) setProfileMappings(profile *PGProfile) (*ProfileEntry, *PKIDEntry) {
```

Comment:
Again, I think the casting from PGEntry to ProfileEntry shoudl happen before we call set*Mappings in order to maintain a clean interface and keep Postgres as low in teh stack as possible.